### ISSUE TITLE
Identifying cases were prepInputs fails to correctly download from URL when manual works

### ISSUE THREAD

**Issue by araymund83** [PredictiveEcology/reproducible#174]
Implement a mechanism within prepinputs to let the user used a not secure connection to allow downloading data. 
Related to this error
```
Downloading http://sis.agr.gc.ca/cansis/nsdb/ecostrat/district/ecodistrict_shp.zip ...
Error in curl::curl_fetch_disk(url, x$path, handle = handle) : 
  Recv failure: Connection reset by peer
```

**Comment by achubaty** [comment]
How is this an SSL issue? Do you have a reproducible example we can work with?

1. it's not clear if the server is redirecting to https (a good thing, but not necessarily the case here). If it were to, there is nothing we can do about it.

```
curl -v 'http://sis.agr.gc.ca/cansis/nsdb/ecostrat/district/ecodistrict_shp.zip'
*   Trying 192.197.71.59:80...
* TCP_NODELAY set
* Connected to sis.agr.gc.ca (192.197.71.59) port 80 (#0)
> GET /cansis/nsdb/ecostrat/district/ecodistrict_shp.zip HTTP/1.1
> Host: sis.agr.gc.ca
> User-Agent: curl/7.68.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Date: Wed, 18 Nov 2020 23:49:26 GMT
< Server: Apache/2.2.15 (Red Hat)
< Last-Modified: Fri, 20 Jan 2012 17:09:51 GMT
< ETag: "360583-8d92f3-4b6f8bfb075c0"
< Accept-Ranges: bytes
< Content-Length: 9278195
< Connection: close
< Content-Type: application/zip
< 
Warning: Binary output can mess up your terminal. Use "--output -" to tell 
Warning: curl to output it to your terminal anyway, or consider "--output 
Warning: <FILE>" to save to a file.
* Failed writing body (0 != 1099)
* Closing connection 0

curl -I 'http://sis.agr.gc.ca/cansis/nsdb/ecostrat/district/ecodistrict_shp.zip'
HTTP/1.1 200 OK
Date: Wed, 18 Nov 2020 23:52:25 GMT
Server: Apache/2.2.15 (Red Hat)
Last-Modified: Fri, 20 Jan 2012 17:09:51 GMT
ETag: "360583-8d92f3-4b6f8bfb075c0"
Accept-Ranges: bytes
Content-Length: 9278195
Connection: close
Content-Type: application/zip
```

This looks like a server config problem, which we have no control over. Closing.

**Comment by eliotmcintire** [comment]
Ana and I spoke about this today. I suggested creating an issue related to `prepInputs` failing. What needs to be distinguish is:
1. where the failures are a `prepInputs` "decision" to fail when `prepInputs` could "decide" to give user the option to succeed,
vs. 
2. where the failures are "outside of the control of `prepInputs`" 

In case 1 -- `prepInputs` should fail gracefully, giving user option to succeed by things like a) manually go to url `absde.com` and put file in `c:/whatever/folder` or b) detect if it is a ssl problem that can be bypassed, with user permission (interactive or via option)

In case 2 -- e.g., the file is no longer at the url, or there is no way to detect the url, then it can fail.

In at least 3 cases that I know of recently, 3 different users have had `prepInputs` fail, because of some thing that is not case 2.

**Comment by achubaty** [comment]
There are so many reasons why a file may fail to download, we can't possibly account for all of them. We can suggest a few things, but at the end of the day if the url fails, there are so many ways it can fail.

1. check the spelling of thu url;
1. check that the url works in your browser on the machine you are having problems on;
2. check that the url works using `curl` on the machine you are having problems on;
3. check the url from another machine (there could be rate limiting or other IP-based blocks on the server side);
4. check your arguments to `prepInputs` to make sure you are calling the function correctly (most of the problems I've helped debug have been with the function being called with incomplete or incorrect arguments);

Disabling SSL verification is a BAD thing to do.

**Comment by achubaty** [comment]
For Case 1 above, I think 'failing gracefully' is wrong. Providing a better error, sure. But it _should_ error/fail if the file download fails.

**Comment by eliotmcintire** [comment]
I believe that this is a solution that favours the advanced user (e.g., Alex Chubaty), and allows the beginner at least some hope of resolving the problem. I strongly believe that this should be allowed to be in an interactive pause. But for now, it will fail "gracefully", meaning it will give some possible hope for a user to recover. Here is an example stop message if a download fails for any reason:
```
Error in downloadFile(archive = archive, targetFile = targetFile, neededFiles = neededFiles,  : 
  Download of rasterTest from https://github.com/tati-micheletti/host/raw/master/data/rasterTest failed; Perhaps try manually downloading it. In your browser, go to
https://github.com/tati-micheletti/host/raw/master/data/rasterTest
, download it manually, give it this name: 'rasterTest', and place file here: C:/Eliot/data. You may also need to run Checksums manually after you download the file: 
-------------------
reproducible:::appendChecksumsTable(checkSumFilePath = 'C:/Users/emcintir.W-VIC-A144916/AppData/Local/Temp/Rtmp04xEwI/Require/obnZhV/CHECKSUMS.txt', filesToChecksum = 'rasterTest', destinationPath = 'C:/Users/emcintir.W-VIC-A144916/AppData/Local/Temp/Rtmp04xEwI/Require/obnZhV', append = TRUE)
```

**Comment by achubaty** [comment]
resolved in #175
